src/plugins/discover/public/components/discover_grid/discover_grid.tsx

```diff
@@ -416,14 +426,49 @@ export const DiscoverGrid = ({
     [dataView, onFieldEdited, services.dataViewFieldEditor]
   );
 
+  const visibleColumns = useMemo(
+    () => getVisibleColumns(displayedColumns, dataView, showTimeCol) as string[],
+    [dataView, displayedColumns, showTimeCol]
+  );
+
+  const getCellValue = useCallback<UseDataGridColumnsCellActionsProps['getCellValue']>(
+    (fieldName, rowIndex) => {
+      const pageRowIndex = rowIndex % displayedRows.length;
+      return displayedRows[pageRowIndex].flattened[fieldName] as CellActionValue;
+    },
+    [displayedRows]
+  );
+
+  const fields = useMemo<UseDataGridColumnsCellActionsProps['fields']>(
+    () =>
+      cellActionsTriggerId
+        ? visibleColumns.map((columnName) => {
+            const field = dataView.getFieldByName(columnName)?.spec;
```

Comment:
Discover works in 2 modes: a regular (simply for data views) one and for text-based search (for example SQL). And users can enable SQL via `discover:enableSql` Advanced Setting and then also save a search on Discover as a saved search which can be added to a dashboard. 

Retrieving a field from a data view will work for the regular mode but unfortunately not for text based mode as columns can be named by user via custom variables. Information about columns is returned by Elasticsearch in a different format and it can be accessed in `discover_documents.tsx` under `documentState.textBasedQueryColumns` if necessary. `isPlainRecord` variable indicates that the current mode is text-based.

Not sure though if it makes sense in Security Solutions to enable special cell actions for text-based searches. If not, then we could return `undefined` fields here if  `isPlainRecord` like the following:
```
() => cellActionsTriggerId && !isPlainRecord ? visibleColumns.map... : undefined
```